rust-runtime/aws-smithy-types/src/error.rs

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Generic errors for Smithy codegen
+
+use crate::retry::{ErrorKind, ProvideErrorKind};
+use std::collections::HashMap;
+use std::fmt;
+use std::fmt::{Display, Formatter};
+
+pub mod display;
+
+/// Generic Error type
+///
+/// For many services, Errors are modeled. However, many services only partially model errors or don't
+/// model errors at all. In these cases, the SDK will return this generic error type to expose the
+/// `code`, `message` and `request_id`.
+#[derive(Debug, Eq, PartialEq, Default, Clone)]
+pub struct Error {
+    code: Option<String>,
```

Comment:
These _could_ be `Cow<'static, str>` rather than `String` and the setters _could_ accept `impl Into<Cow<'static, str>>`.

Reply:
Realistically, none of these fields will be `'static` since they'll always be coming from the body of a HTTP response.